### DISCUSSION TITLE
How do I distinguish between mobile and desktop environments

### DISCUSSION THREAD

**Discussion by j4587698** [AvaloniaUI/Avalonia#9850] [Q&A]
OSPlatform only have WIndows, Linux, Mac.How to judge the mobile terminal?
Like OSPlatform.Mobile or OSPlatform.Android, OSPlatform.IOS?

**Comment by robloo** [comment]
Using Avalonia 11.0:

```csharp
RuntimePlatformInfo? runtimeInfo = AvaloniaLocator.Current.GetService<IRuntimePlatform>()?.GetRuntimeInfo();
bool isMobile = runtimeInfo?.IsMobile ?? false;
```

You can easily get the following platform information (UI example from the Control Catalog):

![image](https://user-images.githubusercontent.com/17993847/210685567-df7297fe-3543-483c-862b-ad8615ce53a1.png)

**Comment by maxkatz6** [comment]
> Like OSPlatform.Mobile or OSPlatform.Android, OSPlatform.IOS?

Use default .NET APIs: OperatingSystem.IsIOS or OperatingSystem.IsAndroid
See https://learn.microsoft.com/en-us/dotnet/api/system.operatingsystem?view=net-7.0

That build in .NET API will cover almost every possible use case for you.
But if you want to know if current device is actually mobile (not some specific OS), even if it's running from the browser (where OS == Browser and not android or ios), they you can use what @robloo suggested in https://github.com/AvaloniaUI/Avalonia/discussions/9850#discussioncomment-4597167

**Reply by danwalmsley** [reply]
OnFormFactor

**Comment by beto-rodriguez** [comment]
Currently AvaloniaLocator.Current is not accessible; how can I determine if the device is mobile?

**Reply by kekekeks** [reply]
`RuntimeInformation.IsOSPlatform(new OSPlatform("Android")) || RuntimeInformation.IsOSPlatform(new OSPlatform("iOS")) ` should work.

cc: @maxkatz6

**Reply by maxkatz6** [reply]
Don't use RuntimeInformation API. Use OperatingSystem.IsAndroid() API

**Reply by beto-rodriguez** [reply]
Thanks that works!

**Reply by robloo** [reply]
The same was done I my end. If you look at the source for RuntimeInfirmation.IsMobile it is just checking if the OS is Android or iOS. That said, this is clearly not correct in all cases. I'm not sure anyone has worked out the logic to differentiate chrome OS and browser running on Android, desktop, etc yet though. Most apps instead use pixel size to switch views to mobile.
